jitter/github.nim

```diff
@@ -0,0 +1,126 @@
+import std/[httpclient, strformat, strutils, json, uri, os]
+
+import extract, parse, log
+
+#TODO add support for appimage downloads
+#TODO prefer appimages -> .tar.gz -> .tgz -> .zip
+
+let baseDir = getHomeDir() & ".jitter/"
+let dDir = baseDir & "nerve/"
+
+proc listPkgGhReleases*(pkg: Package): seq[string] = 
+  ## List and return pkg release tags.
+  let url = fmt"https://api.github.com/repos/{pkg.owner}/{pkg.repo}/releases"
+  let client = newHttpClient()
+  var content: string
+
+  try:
+    content = client.getContent(url)
+  except HttpRequestError:
+    fatal "Failed to fetch repository"
+  finally:
+    client.close()
+
+  let data = content.parseJson()
+
+  if data.kind != JArray:
+    fatal fmt"Failed to fetch {pkg.gitFormat} releases"
+
+  info fmt"Listing release tags for {pkg.gitFormat}"
+
+  for release in data.getElems():
+    list release["tag_name"].getStr()
+    result.add(release["tag_name"].getStr())
+
+proc searchGhRepo*(repo: string): seq[Package] = 
+  let url = "https://api.github.com/search/repositories?" & encodeQuery({"q": repo})
+  let client = newHttpClient()
+  var content: string
+
+  try:
+    content = client.getContent(url)
+  except HttpRequestError:
+    fatal "Failed to fetch repositories"
+  finally:
+    client.close()
+
+  let data = content.parseJson()
+  info &"Found {data[\"total_count\"]} repositories matching the query"
+
+  for repo in data["items"]:
+    list &"Github: {repo[\"full_name\"].getStr()}"
+    result.add(parsePkg(repo["full_name"].getStr()).pkg)
+
+proc downloadRelease(pkg: Package, make = true) =
+  let url = 
+    if pkg.tag == "":
+      fmt"https://api.github.com/repos/{pkg.owner}/{pkg.repo}/releases/latest"
+    else:
+      fmt"https://api.github.com/repos/{pkg.owner}/{pkg.repo}/releases/tags/{pkg.tag}"
+
+  let client = newHttpClient(headers = newHttpHeaders([("accept", "application/vnd.github+json")]))
```

Comment:
make this the global client and remove others

Reply:
That would create a client when `github.nim` is imported, that doesn't make any sense.